### DISCUSSION TITLE
unrelated x-data breaks logic

### DISCUSSION THREAD

**Discussion by flappix** [alpinejs/alpine#3457] [7. Bugs (archived)] [answered]
Consider the following code which filters a list of names based on user input

```
<div x-data="{input: ''}">
  <input x-model="input" />
  <div x-data="{users: ['peter', 'bob', 'maria']}">
    <template x-for="user in users.filter ( (u) => u.includes (input) )">
      <li x-html="user"></li>
    </template>
  </div>
</div>
```
[https://jsfiddle.net/wxtckobp/1/](https://jsfiddle.net/wxtckobp/1/)

Works as expected.  
But as soon as I add x-data to the li tag the logic breaks. E.g. try enter the letter ```o``` in the textfield. In the example above this only shows the name ```bob```. But in the example below this shows only the name ```peter``` which does not contain any o. ```bob``` is not shown anymore.

```
<div x-data="{input: ''}">
  <input x-model="input" />
  <div x-data="{users: ['peter', 'bob', 'maria']}">
    <template x-for="user in users.filter ( (u) => u.includes (input) )">
      <li x-data="{useless: 1}" x-html="user"></li>
    </template>
  </div>
</div>
```
[https://jsfiddle.net/wxtckobp/](https://jsfiddle.net/wxtckobp/)

**Comment by ekwoka** [comment, marked as answer]
add an `:key` property on the template with the x-for, likely the value itself here.

This is discussed in #3447
